### PR TITLE
boards: Add support for the Adafruit Feather M0 Basic Proto.

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
The Feather M0 Basic Proto is an Atmel SAMD21 based development board
with USB, UART, SPI, LED and onboard lithium polymer battery charger.

Use of the Adafruit stock photo (adafruit_feather_m0_basic_proto.jpg)
for the board documentation was approved by Adafruit Industries.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
Thanks for the documentation!

File: boards/arm/adafruit_feather_m0_basic_proto/doc/adafruit_feather_m0_basic_proto.rst [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+The SAMD21 MCU has 6 SERCOM based USARTs.  On the Adafruit Feather M0
+Basic Proto, SERCOM0 is the Zephyr console and is available on pins 0
+(RX) and 1 (TX).
+
+SPI Port
+========
+
+The SAMD21 MCU has 6 SERCOM based SPIs.  On the Adafruit Feather M0
+Basic Proto, SPI4 is available on pin 22 (MISO), pin 23 (MOSI), and
```
Comment (nzmichaelh) [inline boards/arm/adafruit_feather_m0_basic_proto/doc/adafruit_feather_m0_basic_proto.rst]: Are these pins correct?  The schematic shows SERCOM4.2/MOSI to pin 5.

**Review by nzmichaelh — APPROVED** [review]
Hi Henrik.  This is a very well done PR, thanks!

File: boards/arm/adafruit_feather_m0_basic_proto/doc/adafruit_feather_m0_basic_proto.rst (hunk 1 above)
Comment (henrikbrixandersen) [reply boards/arm/adafruit_feather_m0_basic_proto/doc/adafruit_feather_m0_basic_proto.rst]: @nzmichaelh Thank you. Adafruit are using a confusing number of different pin numbering schemes on these boards (as seen on the graphic on https://learn.adafruit.com/adafruit-feather-m0-basic-proto/pinouts).

I have used the pin numbers used in the official Adafruit board documentation (scroll past the graphic on the above link for the list).

**Pull request merged**